impl/src/main/java/org/jboss/cdi/tck/tests/definition/bean/BeanDefinitionTest.java

```diff
@@ -130,7 +130,12 @@ public void testRawBeanTypes() {
         assertTrue(bean.getTypes().contains(MyRawBean.class));
         assertTrue(bean.getTypes().contains(MyBean.class));
         assertTrue(bean.getTypes().contains(MyInterface.class));
-        assertTrue(bean.getTypes().contains(MySuperInterface.class));
+
+        // CDI isn't clear whereas we should get MySuperInterface with or without Number parameter.
+        // It assumes JLS must be followed to determine the list of super types
+        // JLS isn't clear itself and Java implementations either. So we need to keep it open and support both
+        // https://bugs.openjdk.org/browse/JDK-8044366
+        assertTrue(containsClass(bean.getTypes(), MySuperInterface.class));
```

Comment:
I understand the `containsClass` helper method is used as a shortcut, and it's used like that elsewhere, so this is probably fine.

I'd just like to point out that, per my understanding of the JLS ambiguity, `MyRawBean` may have either raw `MySuperInterface` or parameterized `MySuperInterface<Number>` as a superinterface type. This test would consider `MySuperInterface<AnythingElse>` correct too. Myself, I'd do something like

```java
assertTrue(bean.getTypes().contains(MySuperInterface.class)
        || bean.getTypes().contains(new TypeLiteral<MySuperInterface<Number>>() {}.getType()));
```

But as mentioned above, the `containsClass` shortcut is used elsewhere too, and it's unlikely CDI implementations would construct `MySuperInterface<SomethingElse>` in this case, so I guess we can keep it as is.

Reply:
> This test would consider MySuperInterface<AnythingElse> correct too. Myself, I'd do something like

That's a good point. For the sake of clarity we should rather use what Ladislav suggests to make it explicit what two types are accepted.